src/KernelAbstractions.jl

```diff
@@ -329,6 +330,14 @@ macro index(locale, args...)
     Expr(:call, GlobalRef(KernelAbstractions, index_function), esc(:__ctx__), map(esc, args)...)
 end
 
+# TODO: Where should we havdle the logic of neutral, adding it to the macro's logic would reduce complexity in terms of using the macro
+# but adding it to the macro may cause some overhead
+macro reduce(op, val, neutral)
```

Comment:
Is the question here if we should auto-detect the neutral element for op + Type?

I think this is fine, expecting the user to provide their own neutral element.

Reply:
Yes, that was the question.